plugins/micron/micron-nvme.c

```diff
@@ -243,7 +234,7 @@ static int SetupDebugDataDirectories(char *strSN, char *strFilePath,
 			fileLocation = (char *)malloc(length + 1);
 			if (!fileLocation)
 				goto exit_status;
-			strncpy(fileLocation, strFilePath, length);
+			strscpy(fileLocation, strFilePath, sizeof(fileLocation));
```

Comment:
Not sure you want to use `sizeof(fileLocation)` here since this is the size of the pointer and not the size of the memory pointed to by the pointer.